controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -58,7 +58,7 @@ import (
 // +kubebuilder:rbac:groups=core,resources=events,verbs=get;list;watch;create;patch
 // +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;patch
 // +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io;bootstrap.cluster.x-k8s.io;controlplane.cluster.x-k8s.io,resources=*,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=clusters;clusters/status;machines;machines/status,verbs=get;list;watch
+// +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=clusters;clusters/status;machines;machines/status,verbs=get;list;watch;create
```

Comment:
Do we need create on clusters or just machines?

Reply:
Was wondering from the previous PR on this, we need patch instead of create?

Reply:
@vincepri only on machines as the KubeadmControlPlane controller should not be creating cluster objects

@randomvariable hmm, create was the only permission that i saw being complained about, but we will need delete as soon as delete/scale down gets implemented. What would this controller be patching on the machine? It updates the machine template at runtime without saving it to the apiserver so that doesn't need a patch, I didn't see anything else modifying the machines, but I could be wrong.